src/App.tsx

```diff
@@ -14,11 +15,142 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type Props = {};
 
-export default App;
+type State = {
+  goods: string[],
+  isVisible: boolean,
+  isReverse: boolean,
+  sortBy: string,
+  goodsLength: number,
+};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    goods: goodsFromServer,
+    isVisible: false,
+    isReverse: false,
+    sortBy: '',
+    goodsLength: 1,
+  };
+
+  showList = () => {
+    this.setState(state => ({ isVisible: !state.isVisible }));
+  };
+
+  reverseList = () => {
+    this.setState(state => ({ isReverse: !state.isReverse }));
+  };
+
+  sortyByName = () => {
+    this.setState({ sortBy: 'alphabet' });
+  };
+
+  sortyByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
+
+  reset = () => {
+    this.setState(({
+      sortBy: '',
+      isReverse: false,
+      goodsLength: 1,
+    }));
+  };
+
+  lengthGetting = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({ goodsLength: Number(event.currentTarget.value) });
+  };
+
+  render() {
+    const {
+      goods,
+      isVisible,
+      isReverse,
+      sortBy,
+      goodsLength,
+    } = this.state;
+
+    let goodsSelected = goods.filter(good => good.length >= goodsLength);
```

Comment:
создай метод и вынеси всю логику туда
```suggestion
    const goodsSelected = this.getVisibleGoods();
```